Exploration_SampleEntropy_IMU.py

```diff
@@ -0,0 +1,448 @@
+        theta[0] = arctan2(R[2,1]/cos(theta[1]),R[2,2]/cos(theta[1]))
+    theta = theta*180/np.pi
+    return theta
+
+def rot2gravity(timeseries, HS, MS, GS):
+    """
+    Function to find the rotation of the foot flat acceleration vector to the 
+    defined lab gravity coordinate system.
+
+    Parameters
+    ----------
+    accel_vec : numpy array
+        3x1 vector of the x,y,z acceleration at foot flat
+
+    Returns
+    -------
+    theta : numpy array
+        3x1 vector for the rotation from the foot flat acceleration to the lab
+        coordinate system
+
+    """
+    #timeseries = up
+    
+    MS_frames = 20
+    
+    GS = GS[0:-2]
+    
+    
+    for count,jj in enumerate(GS[:-1]):
+        # Obtain the acceleration and gyroscope from foot contact (HS) to the
+        # subsiquent midstance (MS). MS is needed for linear drift correction
+        
+        #count = 0
+        #jj = GS[count]
+        
+        jjn = GS[count +1]
+        acc_stride_plus = timeseries[HS[jj]:MS[jjn],1:4]
+        gyr_stride_plus = timeseries[HS[jj]:MS[jjn], 4:7]
+        time_stride_plus = timeseries[HS[jj]:MS[jjn],1]
+        
+        
+        # Need the midstance and next foot contact indices
+        MS_idx = MS[jj]-HS[jj]
+        HS_idx = HS[jjn]-HS[jj]
+        
+        Fflat_accel = np.mean(acc_stride_plus[MS_idx:MS_idx+MS_frames,:],axis = 0)/np.mean(np.linalg.norm(acc_stride_plus[MS_idx:MS_idx+MS_frames,:],axis = 1)).T
+        
+        # Rotation to the gravity vector: provides initial contidion for gyro integration
+        thetai = findRotToLab(Fflat_accel)
+        
+        # Integrate and linearly correct the gyroscope
+        theta = cumtrapz(gyr_stride_plus,time_stride_plus,initial=0,axis=0)
+        theta = theta-theta[MS_idx,:] # Midstance should be zero
+        # Linear correction
+        slope = theta[-1,:]/(len(theta)-MS_idx-1)
+        drift = (slope*np.ones([len(theta),3]))*(np.array([range(len(theta)),range(len(theta)),range(len(theta))])).T
+        drift = drift-drift[MS_idx,:]
+        # Gyro integration initial condition
+        if count == 0:
+           theta = theta-thetai
+           thetai_up = thetai
+        else: 
+           theta = theta-(thetai+thetai_up)/2
+           thetai_up = thetai
+        
+        # Convert to radians
+        theta = theta*np.pi/180
+        # Rotate the accelerometer into the lab coordinate system (LCS)
+        
+        # Note: this rotation has been converted to list comprehension for speed purposes
+        acc_stride_plus_LCS = [(np.array([[cos(ang[2]),-sin(ang[2]),0],[sin(ang[2]),cos(ang[2]),0],[0,0,1]])@np.array([[1,0,0],[0,cos(ang[0]),-sin(ang[0])],[0,sin(ang[0]),cos(ang[0])]])@np.array([[cos(ang[1]),0,sin(ang[1])],[0,1,0],[-np.sin(ang[1]),0,cos(ang[1])]])@acc_stride_plus[kk,:].T).T-np.array([9.81,0,0]) for kk,ang in enumerate(theta)]
+        acc_stride_plus_LCS = np.stack(acc_stride_plus_LCS)
```

Comment:
Are you trying to transpose the matrix here? I would use .T (similar to line 264) to be consistent with the code